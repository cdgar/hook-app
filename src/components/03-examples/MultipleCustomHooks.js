import React from 'react';
import { useFetch } from '../../hooks/useFetch';
import {useCounter} from '../../hooks/useCounter';
import '../02-useEffect/effects.css';

const MultipleCustomHooks = () => {
    const {counter, increment} = useCounter(1);
    const {loading, data} = useFetch(`https://breakingbadapi.com/api/quotes/${counter}`);
    const {author, quote}= !!data && data[0];

    return (
        <div>
            <h1>breaking Bad Quotes</h1>
            <hr/>
            {
                loading 
                ?
                    (
                        <div className="alert alert-info text-center">
                            Loading..
                        </div>
                    )
                :
                    (
                        <div>
                            <blockquote className="blockquote">
                                <p >{quote}</p>
                                <footer className="blockquote-footer">{author}</footer>
                            </blockquote>
    
                            <button className="btn btn-primary" onClick={() => increment()}>Next</button>
                        </div>
                    )
            }

        </div>
    )
}
    
export default MultipleCustomHooks;
