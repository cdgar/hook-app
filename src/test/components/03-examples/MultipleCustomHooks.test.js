import React from 'react';
import { shallow } from 'enzyme';
import MultipleCustomHooks from '../../../components/03-examples/MultipleCustomHooks';
import { useFetch } from '../../../hooks/useFetch';
import {useCounter} from '../../../hooks/useCounter';

jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');

describe('RealExempleRef.js', () => {

    beforeEach(() => {
        useCounter.mockReturnValue({
            counter: 15,
            increment: () => {}
        });
    })

    test('should show MultipleCustomHooks', () => {
        useFetch.mockReturnValue({
            data: null, 
            loading: true, 
            error: null
        });
        const wrapper = shallow(<MultipleCustomHooks />);
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should show information', () => {
        useFetch.mockReturnValue({
            data: [{
                author: 'authorXD',
                quote: 'quoteXDD'
            }], 
            loading: false, 
            error: null
        });

        const wrapper = shallow(<MultipleCustomHooks />);
        expect(wrapper.find('.alert').exists()).toBe(false);
        expect(wrapper.find('p').text().trim()).toBe('quoteXDD');
        expect(wrapper.find('footer').text().trim()).toBe('authorXD');
    })

})