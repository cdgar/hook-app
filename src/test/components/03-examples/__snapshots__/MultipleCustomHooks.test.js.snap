// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RealExempleRef.js should show MultipleCustomHooks 1`] = `
<div>
  <h1>
    breaking Bad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading..
  </div>
</div>
`;
